### PR TITLE
🐛 Surveymonkey source: fix gzip compressed http response caching

### LINKED ISSUE #5982: SAT fails for surveymonkey connector
There is exception when trying to perform read operation with "survey" stream for Surveymonkey source connector, version 0.1.0: 
```Traceback (most recent call last):
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serializers/jsonserializer.py", line 19, in serialize
    return json.dumps(cassette_dict, indent=4) + "\n"
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/__init__.py", line 398, in dumps
    return cls(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 298, in encode
    chunks = list(chunks)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 696, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 620, in _iterencode_dict
    yield _encoder(value)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dima/airbyte/airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py", line 103, in read_records
    yield from super().read_records(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/sources/streams/http/http.py", line 319, in read_records
    yield from self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice)
  File "/home/dima/airbyte/airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py", line 159, in parse_response
    yield from child_record
  File "/home/dima/airbyte/airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py", line 103, in read_records
    yield from super().read_records(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 91, in __exit__
    next(self.__finish, None)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 69, in _patch_generator
    cassette._save()
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 331, in _save
    self._persister.save_cassette(self._path, self._as_dict(), serializer=self._serializer)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/persisters/filesystem.py", line 20, in save_cassette
    data = serialize(cassette_dict, serializer)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serialize.py", line 58, in serialize
    return serializer.serialize(data)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serializers/jsonserializer.py", line 21, in serialize
    raise UnicodeDecodeError(
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x72 in position 1: invalid start byteDoes this HTTP interaction contain binary data? If so, use a different serializer (like the yaml serializer) for this request?

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serializers/jsonserializer.py", line 19, in serialize
    return json.dumps(cassette_dict, indent=4) + "\n"
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/__init__.py", line 398, in dumps
    return cls(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 298, in encode
    chunks = list(chunks)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 696, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/simplejson/encoder.py", line 620, in _iterencode_dict
    yield _encoder(value)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 33, in <module>
    launch(source, sys.argv[1:])
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/entrypoint.py", line 125, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/entrypoint.py", line 116, in run
    for message in generator:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/sources/abstract_source.py", line 121, in read
    raise e
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/sources/abstract_source.py", line 112, in read
    yield from self._read_stream(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/sources/abstract_source.py", line 147, in _read_stream
    for record in record_iterator:
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/airbyte_cdk/sources/abstract_source.py", line 192, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/home/dima/airbyte/airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py", line 103, in read_records
    yield from super().read_records(
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 91, in __exit__
    next(self.__finish, None)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 69, in _patch_generator
    cassette._save()
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/cassette.py", line 331, in _save
    self._persister.save_cassette(self._path, self._as_dict(), serializer=self._serializer)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/persisters/filesystem.py", line 20, in save_cassette
    data = serialize(cassette_dict, serializer)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serialize.py", line 58, in serialize
    return serializer.serialize(data)
  File "/home/dima/.pyenv/versions/38/lib/python3.8/site-packages/vcr/serializers/jsonserializer.py", line 21, in serialize
    raise UnicodeDecodeError(
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x72 in position 1: invalid start byteDoes this HTTP interaction contain binary data? If so, use a different serializer (like the yaml serializer) for this request?
```

### PR DESCRIPTION
**Pull request by avida**
## What
Resolves #5982

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/schemas/survey_questions.json [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
     "required": {
-      "type": "string"
+      "type": ["string", "null"]
     },
     "validation": {
-      "type": "string"
+      "type": ["string", "null"]
     },
     "forced_ranking": {
       "type": "boolean"
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/schemas/survey_questions.json:36]: You admit that 4 properties can be `Null`, are you sure that the rest of the properties cannot be `Null `too?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json [hunk 2]
```diff
@@ -11,8 +11,8 @@
         "title": "Start Date",
         "type": "string",
         "description": "The date from which you'd like to replicate the data",
-        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
-        "examples": "2021-01-01T00:00:00Z"
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z?$",
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:14]: For my understanding: in what cases can we miss an error with `^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$`, but validate it with `^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z?$`?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/schemas/survey_questions.json (hunk 1 above)
Comment (avida) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/schemas/survey_questions.json:36]: Rest of the properties  are required for request (or they have default value) hence they would always be present on output

 https://developer.surveymonkey.com/api/v3/#surveys-id-pages-id-questions

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json (hunk 2 above)
Comment (avida) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:14]: Z is not required anymore

**Review by yevhenii-ldv — APPROVED** [review]

File: docs/integrations/sources/surveymonkey.md [hunk 3]
```diff
@@ -57,4 +57,5 @@ Then go to Settings and copy your access token
 
 | Version | Date       | Pull Request | Subject |
 | :------ | :--------  | :-----       | :------ |
-| 0.1.0   | 2021-07-06 | [4097](https://github.com/airbytehq/airbyte/pull/4097) | Initial Release |
\ No newline at end of file
+| 0.1.1   | 2021-09-10 | [](https://github.com/airbytehq/airbyte/pull/) | Fix caching for gzip compressed http response |
```
Comment (Zirochkaa) [inline docs/integrations/sources/surveymonkey.md:60]:
```suggestion
| 0.1.1   | 2021-09-10 | [5983](https://github.com/airbytehq/airbyte/pull/5983) | Fix caching for gzip compressed http response |
```

File: airbyte-integrations/connectors/source-surveymonkey/acceptance-test-config.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ tests:
   #incremental:
   #  - config_path: "secrets/config.json"
   #    configured_catalog_path: "integration_tests/configured_catalog.json"
-      # future_state_path: "integration_tests/abnormal_state.json"
+  # future_state_path: "integration_tests/abnormal_state.json"
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-surveymonkey/acceptance-test-config.yml:21]:
```suggestion
  #    future_state_path: "integration_tests/abnormal_state.json"
```

**Review by Zirochkaa — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-surveymonkey/acceptance-test-config.yml (hunk 4 above)
Comment (avida) [reply airbyte-integrations/connectors/source-surveymonkey/acceptance-test-config.yml]: done

File: docs/integrations/sources/surveymonkey.md (hunk 3 above)
Comment (avida) [reply docs/integrations/sources/surveymonkey.md]: done

**Review by sherifnada — APPROVED** [review]

**Pull request merged**